rmarkdowns/Week_4.Rmd

```diff
@@ -257,10 +257,134 @@ For your challenges this week, create the necessary code chunks below. You can c
+```
+
+```{r}
+inat_obs_sf_2 <-  inat_obs_df_2 %>% 
+  select(longitude, latitude, datetime, common_name, scientific_name, image_url, user_login) %>% 
+  st_as_sf(coords=c("longitude", "latitude"), crs=4326)
+
+dim(inat_obs_sf)
+```
+
+
+
 **Challenge 2: Create a static map using the `ggplot` function of your target area and target genus/species.**
 
+Scatter plot of data points of Taro on Oahu
+```{r}
+ggplot() +
+  geom_point(data = inat_obs_df_2, aes(x = longitude, y = latitude, color = common_name, text = common_name)) 
+```
+
+Getting Boundary Box
+```{r}
+oahu_bb <- getbb("Oahu")
+oahu_bb 
+```
+
+Retreaving data on the different parts of Oahu
+```{r}
+# retrieving data of streets on Oahu
+oahu_streets <- oahu_bb %>%
+  opq() %>%
+  add_osm_feature("highway", c("motorway", "primary", "secondary", "tertiary")) %>%
+  osmdata_sf()
+
+# retrieving data of small streets on Oahu
+oahu_small_streets <- oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "highway", value = c("residential", "living_street", "unclassified", "service", "footway")) %>%
+  osmdata_sf()
+
+# retrieving data of coastline on Oahu
+oahu_coast <- oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "natural", value = "coastline") %>%
+  osmdata_sf()
+
+
+# retrieving data of national park on Oahu
+oahu_np <- oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "boundary", value = "national_park") %>%
+  osmdata_sf()
+
+# retrieving data of protected areas on Oahu
+oahu_protected <- oahu_bb %>%
+  opq() %>%
+  add_osm_feature(key = "boundary", value = "protected_area") %>%
+  osmdata_sf()
+```
+
+Creating features of the map
+```{r}
+# visualising all retrieved features over each other to form a map of Maui
+p2 <- ggplot() +
+  geom_sf(data = oahu_streets$osm_lines, inherit.aes = FALSE, color = "#ffbe7f", size = .4, alpha = .8) +
+  geom_sf(data = oahu_small_streets$osm_lines, inherit.aes = FALSE, color = "#a6a6a6", size = .2, alpha = .8) +
+  geom_sf(data = oahu_coast$osm_lines, inherit.aes = FALSE, color = "black", size = .8, alpha = .5) +
+  geom_sf(data = oahu_np$osm_polygons, inherit.aes = FALSE, color = "brown", size = .2, alpha = .8) +
+  geom_sf(data = oahu_protected$osm_polygons, inherit.aes = FALSE, color = "green", size = .2, alpha = .8) +
+  geom_point(data = inat_obs_df_2, aes(x = longitude, y = latitude, color = common_name, text = common_name)) + # here is our iNaturalist data
+  geom_sf_text(size = 1, data = oahu_protected$osm_polygons, aes(label = name)) + #here we are adding some labels for our protected areas for context 
+  coord_sf(xlim = c(-158.28058, -157.64863), ylim = c(21.25482, 21.71201), expand = TRUE) + # setting the limits of our map based on the lat/long we got from our OSM bounding box
+  ggtitle("Taro on Oahu", subtitle = "Based on iNaturalist Data as of October 2024") +
+  theme_bw() +
+  labs(
+    color="Common Name", #this changes the title of our legend
+    x = "Longitude",
+    y = "Latitude"
+  ) 
```

Comment:
This is great! Super interesting to see that cluster in the Round Top forest reserve. I'm guessing there are a lot of hikers using iNat in that area :)